api/src/main/java/org/openmrs/PatientProgram.java

```diff
@@ -10,18 +10,23 @@
 package org.openmrs;
 
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.Date;
 import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
 
+import org.openmrs.attribute.Attribute;
+import org.openmrs.customdatatype.CustomValueDescriptor;
+import org.openmrs.customdatatype.Customizable;
 import org.openmrs.util.OpenmrsUtil;
 
 /**
  * PatientProgram
  */
-public class PatientProgram extends BaseOpenmrsData {
+public class PatientProgram extends BaseOpenmrsData implements Customizable<PatientProgramAttribute> {
```

Comment:
Should actually extend BaseCustomizableMetadata which will reduce the code changes you made

Reply:
Sorry, meant to say BaseCustomizableData, I've updated the comment above to say BaseCustomizableData